### PR TITLE
Return positions for creating a snapshot on the follower

### LINKED ISSUE #7511: StreamProcessor should return ReplayStateMachine positions if run on cont. Replay mode
**Description**

If the StreamProcessor runs in a continuously Replay mode then the call to `lastProcessedPosition` should return the `lastSourceRecordPosition` and the  call to `lastWrittenPosition` should return the `lastAppliedEvent` from the ReplayStateMachine 

Similar as it was done in the POC https://github.com/camunda-cloud/zeebe/commit/5478a3efa83ecb324b647bd2934f680bec1605ee 

This is necessary to create valid snapshots for the Followers. 

Blocks https://github.com/camunda-cloud/zeebe/issues/7510

### PR DESCRIPTION
**Pull request by saig0**
## Description

* get last processed and last written position of the stream processor on replay
* use the source event position of the last replayed event as the last processed position
* use the position of the last replayed event as the last written position

## Related issues

closes #7511 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: engine/src/test/java/io/camunda/zeebe/engine/processing/streamprocessor/StreamProcessorReplayModeTest.java [hunk 1]
```diff
@@ -106,6 +107,38 @@ public void shouldReplayContinuously() {
...
+
+    // when
+    final var commandPosition = replayContinuously.writeCommand(ACTIVATE_ELEMENT, RECORD);
+    final var eventPosition =
+        replayContinuously.writeEvent(
+            ELEMENT_ACTIVATING, RECORD, event -> event.sourceRecordPosition(commandPosition));
+
+    // then
+    final InOrder inOrder = inOrder(typedRecordProcessor, eventApplier);
+    inOrder.verify(eventApplier, TIMEOUT).applyState(anyLong(), eq(ELEMENT_ACTIVATING), any());
```
Comment (deepthidevaki) [inline engine/src/test/java/io/camunda/zeebe/engine/processing/streamprocessor/StreamProcessorReplayModeTest.java:123]: Here you don't need inorder. You can use Mockito.verify.

**Review by deepthidevaki — CHANGES_REQUESTED** [review]
Thanks. Looks good. Just a minor comment.

**Comment by saig0** [comment]
@deepthidevaki good point :sweat_smile: 
I applied your hint. Please have a second look :cookie:

**Review by deepthidevaki — APPROVED** [review]
:tada:

**Pull request merged**